draft-salowey-wimse-arch.md

```diff
@@ -121,11 +125,21 @@ Typically a workload obtains its identity early in its lifecycle. This identity
 * Local API - the identity is provided through an api such as a local domain socket (SPIFFE) or local network API calls (Cloud Provider Metadata Server).
 * Environment Variables - identity may also be injected into workloads using operating system environment variables.
 
+## Agent
+
+The Agent performs the function of translating the underlying Security Context and Workload Identity into Workload Identity Credentials.  The Agent makes the Identity Credentials available to the Workload.
```

Comment:
This makes it sound like the security context closely bound to the workload identity credentials and is obtained in the same way through an agent.  I was thinking that the security context is a bit more loosely coupled with the identity, for example a token containing context information such as user name could reference the workload identity.  It may be delivered in line with a request instead of through an "agent".  Do you see the agent taking a broader role here or are we thinking of different things?

Reply:
Good point.  I was trying to explain the role of the Agent contacting the Server for Identity credentials.  Perhaps this doesn't match what you had in mind.   I've proposed some new wording for the Agent description. 

The Server takes the role of issuing Bearer tokens or X.509 certificates to the Workload.  I was a little confused that there was no Attestation function performed by the Server in the diagram before my PR.   I think the Agent would be responsible for obtaining Workload credentials from the Server.  The "Host Operating System and Hardware" could provide the Agent with locally-signed Agent-credentials that are not useful for Workload to Workload communication.  The Agent could use these credentials when contacting the Server with the Workload Identity, to obtain Workload Identity Credentials suitable for Workload to Workload communication.  There may or may not be a security perimeter between the Workload and the Agent.